ui/app/src/model/datasource-api.ts

```diff
@@ -25,26 +25,30 @@ export function useDatasourceApi(): DatasourceStoreProviderProps['datasourceApi'
 
       // Just resolve a default PrometheusDatasource right now
       if (selector.kind === 'PrometheusDatasource' && selector.name === undefined) {
+        const name = 'PrometheusDemo';
         return {
-          kind: 'GlobalDatasource',
-          metadata: {
-            name: 'PrometheusDemo',
-            created_at: '',
-            updated_at: '',
-            version: 0,
-          },
-          spec: {
-            default: true,
-            display: {
-              name: 'Prometheus Demo',
+          resource: {
+            kind: 'GlobalDatasource',
+            metadata: {
+              name,
+              created_at: '',
+              updated_at: '',
+              version: 0,
             },
-            plugin: {
-              kind: 'PrometheusDatasource',
-              spec: {
-                direct_url: 'https://prometheus.demo.do.prometheus.io',
+            spec: {
+              default: true,
+              display: {
+                name: 'Prometheus Demo',
+              },
+              plugin: {
+                kind: 'PrometheusDatasource',
+                spec: {
+                  direct_url: 'https://prometheus.demo.do.prometheus.io',
+                },
               },
             },
           },
+          proxyUrl: `/proxy/globaldatasources/${encodeURIComponent(name)}`,
```

Comment:
Yeah, good question. So a couple reasons I structured it this way:

1. I was trying to keep the logic of "how to construct URLs" out of the provider component, so that's why I put it into the API object.
2. Every time we get a global/project datasource, we also need its proxy URL, so I _could_ make this two different function calls, but if I need it every time, made sense to just return it all from one function call.

That being said, there's nothing stopping the API implementation here from sharing logic. If we wanted to create a helper function there like you suggest and have both API functions use it, totally cool.